src/phoenix/trace/filter.py

```diff
@@ -79,8 +79,17 @@ def _allowed_replacements() -> Iterator[Tuple[str, ast.expr]]:
         )
     }.items():
         yield source_segment, ast_replacement
-        yield "attributes." + source_segment, ast_replacement
-        yield "span.attributes." + source_segment, ast_replacement
+
+    for source_segment, ast_replacement in {
+        field_name[len(COMPUTED_PREFIX) :]: _ast_replacement(f"span.attributes.get('{field_name}')")
+        for field_name in (
+            getattr(traces, variable_name)
+            for variable_name in dir(traces)
+            if variable_name.isupper()
```

Comment:
suggestion: I think relying on the casing here is a bit of a level of indirection that potentially has pitfalls where the casing of a variable ends up having downstream consequences that are not intended. If you want to keep this dynamic strategy I think that works since it does scale forward - however what I've found useful is to at least write a test that ensures there are guardrails around the set of fields. Seems a bit strange but this way the next person adding a computed field just has to update 1 test and if someone accidentally adds a field that they don't want made accessible, they need to figure out a remedy.

Reply:
Agree. But the casing is the sole factor here. It's just a prefilter. The real filter is the one below where it finds a string value that starts with the computed prefix. Let me put them into an enum instead then we won't need to have this conversation.